pkg/minishift/docker/image/oci_image_handler.go

```diff
@@ -466,8 +510,12 @@ func (handler *OciImageHandler) progressStatusForError(err error) ProgressStatus
 	return OK
 }
 
-func (handler *OciImageHandler) updateIndex(config *ImageCacheConfig, index *Index) error {
-	indexPath := filepath.Join(config.HostCacheDir, "index.json")
+func (handler *OciImageHandler) updateIndex(cacheDir string, index *Index) error {
+	// We only need lock for file write operation.
+	// Check https://play.golang.org/p/2eQmVdxI1LQ for reference.
+	mu.Lock()
+	defer mu.Unlock()
+	indexPath := filepath.Join(cacheDir, "index.json")
```

Comment:
@praveenkumar I think this will not solve the issue of race condition. Because when two process have the same copy of `availableImageIndex` from `availableImageIndex, err := handler.getIndex(config.HostCacheDir)` then they are not aware of each other and would have appended the new value which is applicable to the process.  We need to make sure that each process gets the correct copy of `availableImageIndex`. So only one process at one time should get value of `availableImageIndex` -> update -> write it to the disk then other process take control of it. WDYT?

Reply:
@LalatenduMohanty Honestly, I am not able to process it from my mind mostly because I am not able to test it out properly so I will just revert it back and if you think that is really we need to do might be create an followup issue.

Reply:
>  I will just revert it back and if you think that is really we need to do might be create an followup issue.

+1, lets do that.